mapof.go

```diff
@@ -94,6 +95,19 @@ func NewMapOf[K comparable, V any](options ...func(*MapConfig)) *MapOf[K, V] {
 func NewMapOfWithHasher[K comparable, V any](
 	hasher func(K, uint64) uint64,
 	options ...func(*MapConfig),
+) *MapOf[K, V] {
+	cmp := func(a, b K) bool { return a == b }
+	return NewMapOfWithMethods[K, V](hasher, cmp, options...)
+}
+
+// NewMapOfWithHasher creates a new MapOf instance configured with
+// the given hasher and options. The hash function is used instead
+// of the built-in hash function configured when a map is created
+// with the NewMapOf function.
```

Comment:
Would be great to document the purpose of comparator here similar to how it's done for the hasher.